app/services/facets.rb

```diff
@@ -0,0 +1,24 @@
+module Facets
+  module_function
+
+  def registered_facets
+    configuration.dup
+  end
+
+  def register(facet_model, facet_name = nil, &block)
+    entry = Facets::Entry.new(facet_model, facet_name)
+
+    entry.instance_eval(&block) if block_given?
```

Comment:
Anything except very basic initialization is done via the block:

``` Ruby
  Facets.register(PuppetFacet) do
    extend_model PuppetHostExtensions
    add_helper PuppetFacetHelper
    add_tabs :puppet_tabs
    api_view :list => 'api/v2/puppet_facets/base', :single => 'api/v2/puppet_facets/single_host_view'
    template_compatibility_properties :environment_id, :puppet_proxy_id, :puppet_ca_proxy_id
  end
```

As part of the documentation effort, I will add this example too.